src/Mvc/Mvc.DataAnnotations/src/DataAnnotationsMetadataProvider.cs

```diff
@@ -481,7 +481,7 @@ internal static bool HasNullableAttribute(IEnumerable<object> attributes, out bo
             // See: https://github.com/dotnet/roslyn/blob/master/docs/features/nullable-reference-types.md#annotations
             if (nullableAttribute.GetType().GetField(NullableFlagsFieldName) is FieldInfo field &&
                 field.GetValue(nullableAttribute) is byte[] flags &&
-                flags.Length >= 0 &&
+                flags.Length > 0 &&
```

Comment:
There should be a test to cover this, as it seems there is none, as it didn't fail with IndexOutOfRangeException :wink:

Reply:
According to docs Length will always be greater than 0. Unless roslyn team changes their mind 😆.

Reply:
Ah, good to know. So you could remove the condition, and use a `Debug.Assert(flags.Length > 0)` here. I know it's in the `if`, but it would remove one branch in the resulting code. It this it is worth, anyway.
Otherwise keep it as is.

Reply:
I will keep it as is.